### PR TITLE
Clean Up

### PR DESCRIPTION
**Pull request by Buddybenj**
Read commits for details.

### CODE REVIEW

**Comment by Buddybenj** [comment]
Bump. Merge? @delroth

**Comment by Buddybenj** [comment]
I added a new commit. Please review! Also, once this PR has been reviewed, I will squash the commits into one.

**Comment by degasus** [comment]
@dolphin-emu-bot rebuild

**Comment by Buddybenj** [comment]
Fixed the merge conflicts and squashed the commit into one to make it easier. Does anyone mind rebuilding this PR?

**Comment by magumagu** [comment]
@dolphin-emu-bot rebuild

**Comment by phire** [comment]
@dolphin-emu-bot rebuild

**Comment by Sonicadvance1** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DiscIO/WbfsBlob.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ WbfsFileReader::WbfsFileReader(const std::string& filename)
 
 WbfsFileReader::~WbfsFileReader()
 {
-	for (u32 i = 0; i != m_files.size(); ++ i)
+	for (u32 i = 0; i != m_files.size(); ++i)
```
Comment (delroth) [inline Source/Core/DiscIO/WbfsBlob.cpp]: for (auto& f : m_files[i]) delete f;

Or better, make that a vector/array of unique_ptrs, if possible.

File: Source/Core/DiscIO/WbfsBlob.cpp [hunk 2]
```diff
@@ -40,10 +40,8 @@ WbfsFileReader::WbfsFileReader(const std::string& filename)
 
 WbfsFileReader::~WbfsFileReader()
 {
-	for (u32 i = 0; i != m_files.size(); ++ i)
-	{
-		delete m_files[i];
-	}
+	for (auto& f : m_files[i])
```
Comment (lioncash) [inline Source/Core/DiscIO/WbfsBlob.cpp]: what

Comment (Buddybenj) [inline Source/Core/DiscIO/WbfsBlob.cpp]: Thanks, fixing...

Comment (delroth) [inline Source/Core/DiscIO/WbfsBlob.cpp]: It's pretty much a perfect use case for auto here — nothing depends on the exact type in that loop.

Comment (lioncash) [inline Source/Core/DiscIO/WbfsBlob.cpp]: Either one works, however, making the type explicit means no one has to check the header to see what is actually being deleted.

Comment (delroth) [inline Source/Core/DiscIO/WbfsBlob.cpp]: But nobody really cares about what's being deleted imo.

I'll let you or @Buddybenj make the call.

File: Source/Core/DiscIO/WbfsBlob.cpp [hunk 3]
```diff
@@ -40,10 +40,8 @@ WbfsFileReader::WbfsFileReader(const std::string& filename)
 
 WbfsFileReader::~WbfsFileReader()
 {
-	for (u32 i = 0; i != m_files.size(); ++ i)
-	{
-		delete m_files[i];
-	}
+	for (file_entry* entry : m_files)
+		delete f;
```
Comment (lioncash) [inline Source/Core/DiscIO/WbfsBlob.cpp]: you have to change the `f` to `entry` otherwise you're trying to delete a non-existing variable.

**Comment by Buddybenj** [comment]
Merge? @lioncash

**Comment by degasus** [comment]
LGTM

**Pull request merged**